File: frontend/src/types/activity.ts
export interface Activity {
  id: number;
  workflowId: number;
  name: string;
  attempt: number;
  success: boolean;
  startedAt: string | number[] | null;
  finishedAt: string | number[] | null;
  inputPayload: string | null;
  outputPayload: string | null;
  errorMessage: string | null;
}

export interface LastActivity {
  name: string;
  attempt: number;
}

File: src/main/java/com/beeline/temporalmini/Activity.java
package com.beeline.temporalmini;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "activity", schema = "wflow")
public class Activity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long workflowId;
    private String name;
    private int attempt;
    private boolean success;
    private LocalDateTime startedAt;
    private LocalDateTime finishedAt;
    @Column(columnDefinition = "text")
    private String inputPayload;
    @Column(columnDefinition = "text")
    private String outputPayload;
    /** Fully-qualified class name of the deserialized {@code outputPayload}; used by replay to reconstruct the typed result. */
    private String outputType;
    private String errorMessage;
}
